frontend/src/components/Year/Year.tsx

```diff
@@ -74,37 +87,55 @@ class YearComponent extends React.Component<Props> {
                 <KeyboardArrowDownIcon />
               ))}
           </div>
-          <YearTop
-            year={year}
-            fallStatus={schedule.yearMap[year].fall.status}
-            springStatus={schedule.yearMap[year].spring.status}
-            summer1Status={schedule.yearMap[year].summer1.status}
-            summer2Status={schedule.yearMap[year].summer2.status}
-            schedule={schedule}
-            isEditable={isEditable}
-            transferCourses={this.props.transferCourses}
-          />
-        </YearTopWrapper>
-        {isExpanded && (
-          <YearBody>
-            <SemesterBlock
+          <SemestersWrapper>
+            <SemesterContainer
```

Comment:
Want to find a way to map through the different semesters throughout the year instead of hardcoding each semester.

Reply:
Hmm, perhaps mapping through the semesters array? It seems like yall were on the path to doing something similar since I saw a semesters array in the `SemesterTop` component.

Something like this seemed to work for me

```tsx
const semesters: SeasonWord[] = ["fall", "spring", "summer1", "summer2"];
{semesters.map(semester => {
              const scheduleTerm = schedule.yearMap[year][semester];
              return (
                <SemesterContainer
                  isEditable={isEditable}
                  semester={scheduleTerm}
                  year={year}
                  semesterStatus={
                    scheduleTerm.status as StatusEnum
                  }
                  schedule={schedule}
                  transferCourses={this.props.transferCourses}
                  semesterWord={semester}
                  isExpanded={isExpanded}
                />
           );
})}
```